### ISSUE TITLE
Scope

### ISSUE THREAD

**Issue by JAForbes** [isiahmeadows/livescript-fork#1]
We should figure out what we're doing and what we're not doing.

**Comment by isiahmeadows** [comment]
I'm working on an issue doing at least part of this.

**Comment by JAForbes** [comment]
I agree with @isiahmeadows  think the first order problem with livescript is compiler issues.

The error message quality isn't great, tooling is not up to modern standards and source maps are iffy.

There's also a bunch of language features I think we could drop, both to simplify the language and also liberate certain operators to be used for other things.

I posted this in the mithril gitter:

```
The gold in the language for me is:

the composition operators/pipeline
operators as functions e.g. (.a.b.c is \f)instead of x => x.a.b.c === 'f'
the way they manage template strings
chained comparisons
keywords for logic operators (e.g. xor )
shorthand switches
expression oriented
\strings
ranges
it / that
The stuff I'd cut:

All the weird literal extensions, commaless arrays, yaml like objects, yes/no aliases etc
partial application, auto currying, unnecessary if you have arrow functions
bound function syntax
any deviation from esm
! invocation
paren less invocation
named functions being different to es5 (aside from auto return)
long form switches
coffescript style switches
comprehensions
do
with
I think I like cascades, but I'd expect an assignment in a cascade to be immutable, not sure, leaning towards cutting
```

[Link](https://gitter.im/mithriljs/mithril.js?at=5c26060b37975e7ca937c345)

I'd recommend we tackle the compiler issues and if a particular feature on the naughty list is creating problems resolving that first order task, we just cut it.

**Comment by isiahmeadows** [comment]
I'll note that many of the compiler issues have been solved if you look [in their changelog](http://livescript.net/#changelog) for v1.6.0. (I do criticize their decision to not up it to v2.0.0, given there's significant breaking changes and new features.)

**Comment by isiahmeadows** [comment]
BTW, I think our big advantages over LiveScript is we know how dynamic functional languages should work (we've tried several), we prefer simplicity, and we'd rather provide the primitives that can build the big utilities.
